### ISSUE TITLE
Fix: Sticky header not blurring with search overlay when stickyNav is enabled

### ISSUE THREAD

**Issue by ZeeL981210** [timlrx/tailwind-nextjs-starter-blog#1036]
**Describe the bug**
Please see the figure below: 
<img width="821" alt="image" src="https://github.com/user-attachments/assets/e204e7e3-b8d8-4554-a645-d0c252a7f70d">
I suppose the header shall be hidden in such case, which the searching section is popped. And also in this case, if you click the search button once-more, all the components are shifted to the left upon each click.

**To Reproduce**
When configure `stickyNav` to `true` in `siteMetadata.js`, it allows the header to stick on top of the page
1. Go to `siteMetadata.js`
2. Set `stickyNav` to true

**Expected behavior**
Normally the header should be blured as shown in the figure below:
<img width="823" alt="image" src="https://github.com/user-attachments/assets/7e390c9b-6606-47a7-acf5-ecbba8004b64">

**Additional context**
I attempted to fix it, but I failed. My best guess is that this issue is related to 
```
 if (siteMetadata.stickyNav) {
    headerClass += ' sticky top-0 z-50'
  }
```
that in `Header.tsx`, because when I modify the `z-index` to smaller than `10`, this specific issue get fixed, but it will cause the code block/media player float over the header.

Hope all above helps.

**Comment by timlrx** [comment]
Yep, it's a little tricky since currently most of the stacking logic with the exception of the sticky bar is handled via stacking context. I think I would have to modify the pliny component to allow letting the user specify their own classname so the relative z-index can be better managed

**Comment by ZeeL981210** [comment]
Thank you for the reply. I think that's a great way to solve the problem.

**Comment by timlrx** [comment]
Fixed in pliny v0.4.0. I decided to add a z-index: `z-50` to the overlay instead.
